ir/state.cpp

```diff
@@ -241,6 +241,20 @@ State::addFnCall(const string &name, vector<StateValue> &&inputs,
     return vector<StateValue>(out_types.size());
   }
 
+  for (auto &v : ptr_inputs) {
+    if (!v.nocapture && v.val.non_poison.isTrue()) {
+      Pointer p(memory, v.val.value);
+      if (!p.isLocal().isTrue())
+        continue;
+      // TODO: this should support more complex cases, such as (c ? bid1: bid2).
+      // Current implementation of extract_possible_local_bids cannot be used
+      // because it causes false positive for an input in gzip.
+      uint64_t short_bid;
+      if (p.getShortBid().isUInt(short_bid))
+        memory.escapeLocal(short_bid);
```

Comment:
I really don't like this logic in this file. It should be moved to memory.cpp, e.g. Memory::escapePtr(StateValue&).
If extract_possible_local_bid is not good enough, but we should have a look into why as well. It only needs to be as good as LLVM's constant folding. (this can go into another PR)

Reply:
The failure comes from a code like this:
```
p = alloca
use p ; but never escape p

q = load * glb ; as p is never escaped, q cannot be p
f(q) ; q cannot be p, but current extract_possible_local_bid cannot catch it
```

To do this, we need to track what 'worst case' is & let extract_possible_local_bid get the worst case rather than simply marking all bits as failing.

I'll work on this in a separate PR.

Reply:
For this PR, I'll reflect your comment.